File: week1/c++/week1_2.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int solution(vector<int> people, int limit)
{

    // 정렬
    sort(people.begin(), people.end());

    int answer = 0; // 구명보트의 갯수

    int left = 0;
    int right = people.size() - 1;

    while (left <= right)
    {
        if (left == right)
        {
            answer++;
            return answer;
        }

        if (people[left] + people[right] > limit)
        {
            answer++;
            right--;
        }
        else
        {
            answer++;
            left++;
            right--;
        }
    }

    return answer;
}

File: week1/java/week1_2.java
package week1;

import java.util.Arrays;

public class week1_2 {
    public static int solution(int[] people, int limit) {
        int answer = 0;
        Arrays.sort(people);

        int i = 0;
        int j = people.length - 1;
        while (true) {
            if(i>=j){
                if(i==j) answer++;
                break;
            }
            if(people[i] + people[j] <= limit){
                answer++;
                i++;
                j--;
            }else{
                answer++;
                j--;
            }
            System.out.println(i);
            System.out.println(j);
            System.out.println(answer);
        }
        return answer;
    }
    public static void main(String[] args){
        int[] arr = {70, 50, 80, 50};
        System.out.println(solution(arr, 100));
    }
}
